server/sonar-web/src/main/js/apps/projects/components/AllProjects.js

```diff
@@ -80,24 +89,38 @@ export default class AllProjects extends React.PureComponent {
 
     if (this.state.query.view === 'leak' || view === 'leak') {
       if (this.state.query.sort) {
-        const sort = parseSorting(this.state.query.sort);
-        if (SORTING_SWITCH[sort.sortValue]) {
-          query.sort = (sort.sortDesc ? '-' : '') + SORTING_SWITCH[sort.sortValue];
+        const sort = utils.parseSorting(this.state.query.sort);
+        if (utils.SORTING_SWITCH[sort.sortValue]) {
+          query.sort = (sort.sortDesc ? '-' : '') + utils.SORTING_SWITCH[sort.sortValue];
         }
       }
       this.props.router.push({ pathname: this.props.location.pathname, query });
     } else {
       this.updateLocationQuery(query);
     }
+
+    utils.saveView(view);
+    utils.saveVisualization(visualization);
```

Comment:
Sort should be saved here too, otherwise if you : 
1. Choose leak perspective
2. Choose a leak sorting option
3. Choose the overall perspective
4. Go to issues, come back to project
5. You have the wrong sort selected and remembered in the localstorage

Or maybe all `saveX` actions should be done in `updateLocationQuery`